### DISCUSSION TITLE
Dependency problem of near-runtime v4.0.0-pre.1 of package near-sdk-sim v3.2.0

### DISCUSSION THREAD

**Discussion by jahcacao** [near/nearcore#10428] [Q&A]
I try to build my project, but I get this dependency error:
"
error: failed to select a version for the requirement `near-vm-runner = "^4.0.0-pre.1"`
candidate versions found which didn't match: 0.19.0, 0.19.0-pre.2, 0.19.0-pre.1, ...
location searched: crates.io index
required by package `near-runtime v4.0.0-pre.1`
    ... which satisfies dependency `near-runtime = "=4.0.0-pre.1"` of package `near-sdk-sim v3.2.0`
    ... which satisfies dependency `near-sdk-sim = "^3.2.0"` 
"

My cargo.toml:
"
[lib]
crate-type = ["cdylib", "rlib"]

[dependencies]
near-sdk = "3.1.0"
serde = "0.9.0-rc2"
near-contract-standards = "3.1.0"
near-vm-runner = { version = "0.19.0" }

[dev-dependencies]
near-sdk-sim = "3.2.0"

[patch.crates-io]
parity-secp256k1 = {git = "https://github.com/paritytech/rust-secp256k1"}
"

Do you have any ideas how to solve this problem?

**Comment by TatyOko28** [comment]
The error occurs because near-sdk-sim v3.2.0 depends on near-runtime v4.0.0-pre.1, which in turn requires near-vm-runner = "^4.0.0-pre.1, but the version 4.0.0-pre.1 of near-vm-runner is not available on crates.io.
Solution: Align your dependency versions or use a Git version for near-sdk-sim and related crates.
